libraries/provider_database_mysql_user.rb

```diff
@@ -128,7 +128,7 @@ def ishash?
           if incorrect_privs
             converge_by "Granting privs for '#{new_resource.username}'@'#{new_resource.host}'" do
               begin
-                repair_sql = "GRANT #{new_resource.privileges.join(',')}"
+                repair_sql = "GRANT #{new_resource.privileges.join(',').tr('_', ' ')}"
```

Comment:
Will probably have to do:
`new_resource.privileges.join(',').tr('_', ' ').gsub('repl ', 'replication ').gsub('create tmp table', 'create temporary tables').gsub('show db', 'show databases')`